plugins/user-authenticators/ldap/src/org/apache/cloudstack/api/command/LdapCreateAccountCmd.java

```diff
@@ -119,6 +131,9 @@ private Long getDomainId() {
 
     @Override
     public void execute() throws ServerApiException {
+        if (getAccountType() == null && getRoleId() == null) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Both account type and role ID are not provided");
```

Comment:
@bhaisaab this check seems more restrictive than the previous version.  Before this change, ``accountType`` was completely optional.  With this change, an script using this API call that did not include ``accountType`` and wouldn't include ``roleId``because it is a new parameter would fail.